src/classifier_sequence.py

```diff
@@ -0,0 +1,100 @@
+import pathlib
+import math
+import numpy as np
+from typing import Dict, Tuple, List
+from tensorflow import keras
+from PIL import Image, ImageOps
+
+
+class ClassifierSequence(keras.utils.Sequence):
+    """A class for sequencing the classifier's input data."""
+
+    category_id_to_name: Dict[int, str]
+    category_name_to_id: Dict[str, int]
+
+    def __init__(self, input_dir: pathlib.Path, batch_size: int,
+                 target_img_dim: int, shuffle: bool = True):
+        """Constructor.
+
+        :param input_dir: directory with input data
+        :param batch_size: the desired size of a batch
+        :param target_img_dim: target dimension (for square icons)
+        :param shuffle: whether the input data is shuffled on epoch end or not.
+        """
+        if not input_dir.exists():
+            raise ValueError("Input directory does not exist.")
```

Comment:
For exceptions like this it is really helpful to add what directory doesn't exist, e.g.
`raise ValueError(f"Input directory does not exist: {input_dir}")`